core/src/services/routing.js

```diff
@@ -264,6 +284,6 @@ export const handleRouteClick = node => {
     navigateTo(link);
   } else {
     const route = RoutingHelpers.buildRoute(node, `/${node.pathSegment}`);
-    navigateTo(route);
+    navigateTo(RoutingHelpers.substituteViewUrl(route, componentData));
```

Comment:
To use the RoutingHelpers.substituteViewUrl is confusing because it is not applied on a view Url but on the main luigi navigation path. It works but it is - confusing - and dangerous because we might change the substituteViewUrl in the future and not think that it (mis)used like this. Please write a dedicated function or just use the GenericHelpers.replaceVars for the dynamic path segments directly.